### ISSUE TITLE
Allow edit of image alt text

### ISSUE THREAD

**Issue by kenara** [bluesky-social/social-app#7454]
### Describe the Feature

It's easy to forget adding alt text to an image. And someone might call your attention to an image you posted that is missing alt text. This feature is not to be confused with the much requested, almost mythical "Edit button" for posts. 

The Bluesky community is making a huge deal about image alt texts. Let's make it a bit easier...

**Comment by kenara** [comment]
I don't know if there will ever be - or should be - an edit button. It's a whole other thing that would allow a poster to change the meaning of their post after it has gained likes, etc. It would enable abuse and that's probably why we don't see edit buttons when we could really use one.

On the other hand, forgetting to add alt text is something I've done and been called out for, and since Bluesky has championed the cause of alt text, allowing it to be added retroactively (by the original poster) would make it that much easier to do what many consider "the right thing."

**Comment by surfdude29** [comment]
> @surfdude29, I don't know if there will ever be - or should be - an edit button. It's a whole other thing that would allow a poster to change the meaning of their post after it has gained likes, etc. It would enable abuse and that's probably why we don't see edit buttons when we could really use one.

It's partly because of those concerns and the need to get the user experience right that we don't have post editing yet (the Bluesky team has also just been really busy working on lots of other stuff). However, post editing is definitely planned; there will be a new roadmap issued soon but for now have a look at what the [2024 roadmap says on this](https://docs.bsky.app/blog/2024-protocol-roadmap#on-the-horizon):

> **Record Versioning (Post Editing):** atproto already supports updating records in repositories: one example is updating bsky profile records. And preparations were made early in the protocol design to support post editing while avoiding misleading edits. Ideally, it would also be possible to (optionally) keep old versions of records around in the repository, and allow referencing and accessing multiple versions of the same record.

**Comment by kenara** [comment]
@surfdude29 I understand alt text to be a simple, unbiased description of the image (why we get 2000 characters for that I don't know) and I've been using galaxy.ai (https://galaxy.ai/ai-alt-text-generator) for that. Maybe Bluesky should offer something like that.

**Comment by surfdude29** [comment]
The alt text limit was increased to 2000 characters so that all of the text in text-heavy screenshots can be included.

A built-in alt text generator using AI is an interesting idea, please see #1636 for that proposal.

**Comment by cooljeanius** [comment]
> Please see [#659](https://github.com/bluesky-social/social-app/issues/659) for the ability to add and edit alt text retroactively.

#6559 too

**Comment by kenara** [comment]
@cooljeanius Thanks. Eventually somebody will do something about this. Let's be honest though, only some people care about alt-text and even those who care might forget to add it, find it too challenging to describe their image or not be willing or interested to go back and approve the changes to an old post. Sure, a whole culture could develop around wikipedianizing bluesky posts missing alt-text - the alt-text police - but ultimately, some AI assisted alt-text generation will probably be the answer.
